mathics_scanner/generate/build_tables.py

```diff
@@ -130,9 +137,7 @@ def compile_tables(data: dict) -> dict:
 
     # All supported named characters dictionary entry
     named_characters = {
-        k: v.get("unicode-equivalent", v.get("wl-unicode"))
-        for k, v in data.items()
-        if "wl-unicode" in v
+        k: v["wl-unicode"] for k, v in data.items() if "wl-unicode" in v
```

Comment:
I don't understand the motivation for this change. 

The former says to me that something is a named character if it appears it has a wl-unicode tagging, and that we want to use the unicode equivalent first, and if that doesn't exist the wl-unicode value. 

The change says that we want the wl-unicode value whether or not there is a unicode equivalent. 

I can probably guess what is in your mind but I'd prefer not to guess.

Reply:
The point is we want to take a string in the internal representation, where you have wl-unicode characters, and you want to replace them by a LaTeX command. So, the key is the wl-unicode, and the value is the amstex code.

Reply:
This is why I think we need both fields (`and`)  available .

Reply:
> The point is we want to take a string in the internal representation, where you have wl-unicode characters,

Yeah, this is what I thought. And this was what I was getting to in an earlier remark about being myopic and going in to some area that isn't an area of expertise without hesitation or solicitation and just blasting away. That then puts  pressure on me to play catch up with what was going on in you mind. (And sloppy commits, vague PR comments and using repositories like a local sandbox doesn't help here.)

I get that you'd like to use some sort of WL-symbol internally in mathics-core for things like an operators or named symbols. However as I mentioned before, mathics scanner is already used in other repositories, in particular mathics-scanner and mathics-pygments. 

Therefore what we need are additional tables for the purpose of symbols which can be used internally by mathics-core.
And the descriptions of what the tables mean also needs to be expanded upon.

Reply:
> > The point is we want to take a string in the internal representation, where you have wl-unicode characters,
> 
> Yeah, this is what I thought. And this was what I was getting to in an earlier remark about being myopic and going in to some area that isn't an area of expertise without hesitation or solicitation and just blasting away. That then puts pressure on me to play catch up with what was going on in your mind via PRs. (And sloppy commits, vague PR comments and using repositories PRs like a local sandbox doesn't help here.)
> 

No @Rocky, it was not my intention to put pressure on you. I put a PR because I thought I had understood the logic of this package. I leave some issues explaining what I think is necessary to make to work on an idea on the mathics-core side, related to something that we have previously discussed (the one of using mathics-scanner to implement the conversion of wl-unicode characters to amstex commands). I do not have any hurry on having this finished, but I wanted to communicate the idea about what is need and how I think we can implement it. That's all.  

> I get that you'd like to use some sort of WL unicode symbol internally in mathics-core for things like an operators or named symbols. However as I mentioned before, mathics scanner is already used in other repositories, in particular mathicsscript and mathics-pygments.
> 

Yep, but what I am proposing does not change the behaviour regarding the last release. Just adds some functionalities and propose to revert a change that breaks mathics-core in some extent.

> Therefore what we need are additional tables for the purpose of symbols which can be used internally by mathics-core. And the descriptions of what the tables mean also needs to be expanded upon.

Yes, one possibility is to generate the new tables on the mathics-core side, and then migrate them to here, if it seems useful.

Reply:
> No @rocky, it was not my intention to put pressure on you.

I didn't say and definitely didn't _think_ there was intention. I never ascribe to maliciousness that which is easily explained by thoughtlessness. I want to bring it up because, it does put pressure. Anything Mathics wise I had thought I might do this week has been scraped to make sure we move forward on the work that was done during the week and before. 

Overall I think it is good intent, and probably most of it is good, and much needed!

> (the one of using mathics-scanner to implement the conversion of wl-unicode characters to amstex commands).

Yes, this I was aware of from the discussions. But please, again, be sensitive to when you are straying out what you well know. And when that happens _stop_ and ask. 

> propose to revert a change that breaks mathics-core in some extent.

This is what I mean by myopic. Did you look at the commit in the revert to see _why_ it was changed? What you consider a revert for some purpose may be a breakage for whatever reason it was changed. 

Past experience  has always been that when this happens means that it needs to be looked at more carefully. 
 
> Yes, one possibility is to generate the new tables on the mathics-core side, and then migrate them to here, if it seems useful.

That probably leads to the fastest iteration in development. But again, one thing this activity has made clear to me is that we need a better description of the tables. h



 I put a PR because I thought I had understood the logic of this package. I leave some issues explaining what I think is necessary to make to work on an idea on the mathics-core side, related to something that we have previously discussed (the one of using mathics-scanner to implement the conversion of wl-unicode characters to amstex commands). I do not have any hurry on having this finished, but I wanted to communicate the idea about what is need and how I think we can implement it. That's all.
> 
> > I get that you'd like to use some sort of WL unicode symbol internally in mathics-core for things like an operators or named symbols. However as I mentioned before, mathics scanner is already used in other repositories, in particular mathicsscript and mathics-pygments.
> 
> Yep, but what I am proposing does not change the behaviour regarding the last release. Just adds some functionalities and propose to revert a change that breaks mathics-core in some extent.
> 
> > Therefore what we need are additional tables for the purpose of symbols which can be used internally by mathics-core. And the descriptions of what the tables mean also needs to be expanded upon.
> 
> Yes, one possibility is to generate the new tables on the mathics-core side, and then migrate them to here, if it seems useful.

Reply:
Here 8a2dc4a298d7afe542c8836ff07a2aeeb8c01575 is where the change in the table of NamedCharacters was introduced. There is no further description in the commit. In any case, it was introduced after the last release, so I would expect that the reversion does not have a conflict / breakages on any of the (currently released versions of the ) other packages.

Reply:
I have now gone over the code in history in detail and looked at the mathicsscript and mathics-pygments. 

Here is what I understand. Input has to allow unicode forms because those are the input devices we currently support. We don't have kinds of keyboards that I am aware of that generate WL unicode symbols. 

A _new_ desire in the mathics-core project is to canonicalize various character symbols into the WL unicode symbol. That's okay and that makes sense. Another equally valid alternative would be to canonicalize to the "standard" unicode symbol. 

What is needed is a canonicalization inside the scanner to one either WL unicode or standard unicode if the other kind of unicode was input. When is canonicalization is to standard unicode, I don't think this is an issue.  I gather there is some reason for preferring WL unicode - what is it? 

If input devices always produced WL unicode or in other words if WL had not defined its own unicode for some symbols that we wouldn't need this extra step. 

But when you want to support both kinds of input then _a new step_ is needed to do the conversion when the other form is used. 

`k: v.get("unicode-equivalent", v.get("wl-unicode"))` is there to support unicode input and when that doesn't appear because it is the same as wl-unicode, then we pick it up from wl-unicode. 

The basis for [this comment](https://github.com/Mathics3/mathics-scanner/blob/master/mathics_scanner/data/named-characters.yml#L35)  about not adding a unicode equivalent if it is not needed, that is, either the same thing as the ascii or the same thing as the wl-unicode is related to this. That comment should be elaborated upon, but I'll do that later. 

If that were followed (and the https://github.com/Mathics3/mathics-scanner/pull/45 violates that; we should fix this add a test for this for improper use of unicode-equivalent), then the scanner would convert to wl-unicode only when there is a unicode-equivalent field.  Here a new table for this should be created.